### ISSUE TITLE
How can forwarding be referentially transparent?

### ISSUE THREAD

**Issue by ccorcos** [paldepind/flyd#95]
The issue I'm trying to solve is tying to make this listOf function more performant:

https://github.com/paldepind/functional-frontend-architecture/blob/master/examples/nesting/js/list.js#L38

We can use thunks and stuff to lazily evaluate the UI. But the action$ is going to have a new reference every time so lazily evaluating with the listOf component is helpless. However, what if there was some way of computing `stream.eq(prevAction$, nextAction$)`? We could feasibly do this by keeping track of the forwarding going on.

Maybe I'm barking up the wrong tree, but its frustrating that this list componnt  is going to cause all items to rerender regardless.

**Comment by c-dante** [comment]
If I'm understanding this correctly, the issue is that your update function is just re-generate the entire list on each action, regardless of if it actually changed anything. Is that correct?

**Comment by ccorcos** [comment]
Yes. Well mostly. The thing is the function reference changed, but its still the same stream and the same forwarding. So its effectively *equal* even though it isnt referentially equal. [Here's an example of what I'm thinking](https://github.com/ccorcos/elmish/blob/147ac1730a6fe700cae85db478e7d651b6073730/lazy/main.js).
